### ISSUE TITLE
Mobs randomly despawn instantly

### ISSUE THREAD

**Issue by CW-UK** [Nathat23/StackMob-5#133]
Since the latest update, I've had issues with mobs despawning straight after spawning from mob spawners. It's as though they're stacking into a stack that doesn't exist anywhere. No errors in logs, running Paper 1.16.2 (rev. 188). Friendly mobs are also not persisting after chunk unloads or server restarts, and I'm not running any kind of anti-lag or entity management plugins.

[https://youtu.be/i6Rf19iNpLI](https://youtu.be/i6Rf19iNpLI)
[https://youtu.be/sfI6LTum-L8](https://youtu.be/sfI6LTum-L8)

I'm not sure if this is a StackMob issue or a Paper issue, but I've had no problems up until the last update.

**Comment by Nathat23** [comment]
Is there any chance you have used the '/sm remove' command?

**Comment by CW-UK** [comment]
I haven't, no. All the mobs have either spawned from spawners with no other input or interaction, or have had their stack size increased with the /sm tool. I initially discovered this yesterday when a player's sheep grinder suddenly lost its sheep. All of the stacks of coloured sheep were increased with the tool, however the white/gray/black/brown were all spawned purely by the spawner.

**Comment by Nathat23** [comment]
If you use the following build, https://drive.google.com/file/d/1BavIFDheWjpS9XlQHhJ6ZdHQ93Q08RvC/view?usp=sharing and run the command '/sm remove' does that fix the issue?

**Comment by CW-UK** [comment]
I can try it, but I may not be able to do it until tomorrow as I don't currently have any servers available for testing on. I'll update once I've managed to test it. Is it possible that '/sm remove animals' will fix the issue in the current build I'm using for the meantime? I don't want to try anything if it may just add to my problems 😄

**Comment by Nathat23** [comment]
The build I linked fixes the 'sm remove' command.

**Comment by CW-UK** [comment]
Ah okay, so I need to use that version then run 'remove' to reset everything? No worries, will try and squeeze in an unscheduled restart shortly.

**Comment by CW-UK** [comment]
Loaded this plugin earlier today and all working fine for the stacks randomly disappearing, however with this version, bees aren't stacking, and nor are striders (that I've found so far). Everything else seems to be stacking fine.

**Comment by CW-UK** [comment]
This issue is occurring again with the latest version, and using /sm remove. Seems to be affecting iron golems and sheared sheep.

Iron Golem's will just disappear after spawning, and if I use /sm spawn to bring a stack in, one will die and then the rest of the stack disappears with it, it doesn't just reduce. As for sheep, if I shear them I end up with about half the amount I started with at the end.

**Comment by Nathat23** [comment]
Any errors?

**Comment by CW-UK** [comment]
Nope, nothing at all in logs unfortunately. They just vanish the same as before:

[https://www.youtube.com/watch?v=qgknVopN8RU](https://www.youtube.com/watch?v=qgknVopN8RU)
